experimental/dds/tree2/src/codec/codec.ts

```diff
@@ -69,11 +69,12 @@ export interface ICodecOptions {
 
 /**
  * @alpha
+ * @remarks - `TEncoded` should always be valid Json (i.e. not contain functions), but due to Typescript's handling
```

Comment:
It's not clear to me how this change makes the code that use this interface simpler to write.

Reply:
basically, if you wanted to declare a concrete type for your encoded object, the `extends JsonCompatibleReadOnly` constraint would force you to put an index signature on that type. That isn't great--we don't want to accept arbitrary properties at the type system level.